datastore.go

```diff
@@ -174,6 +174,14 @@ func (a *accessor) queryOrig(q dsq.Query) (dsq.Results, error) {
 		qr = dsq.NaiveFilter(qr, f)
 	}
 	for _, o := range q.Orders {
+		// leveldb iterators return entries sorted lexicographically by key, so we can bypass the naive ordering
+		// for OrderByKey. Since OrderByKey.Sort() has as struct receiver, the value stored in q.Orders can be a
+		// pointer or a struct, so we need to check for both for correctness.
+		if _, ok := o.(dsq.OrderByKey); ok {
```

Comment:
Nit:

```go
switch o.(type) {
case dsq.OrderByKey, *dsq.OrderByKey:
  continue
}
```

Reply:
Changed here and in https://github.com/ipfs/go-ds-badger/pull/43 ;-)